parallel-task-set/src/lib.rs

```diff
@@ -60,6 +61,41 @@ impl<T: 'static + Send> ParallelTaskSet<T> {
         Self { semaphore, set }
     }
 
+    /// Spawn the provided `next_future`, potentially waiting until a previously
+    /// spawned task completes if the `ParallelTaskSet` is at its concurrency
+    /// limit. Note that, *unlike* [`join_next()`], this method returning `None`
+    /// does NOT indicate that the set is empty, just that it was not necessary
+    /// to wait for a pervious task to complete in order to spawn the existing
+    /// one.
+    pub async fn spawn_and_join<F>(&mut self, future: F) -> Option<T>
```

Comment:
As a naming nitpick, this should probably be `join_and_spawn` or just `spawn`, and we rename/deprecate the other flavor

Reply:
I just made this be `spawn` and got rid of the old `spawn`, it turned out to be pretty trivial: abb5284d82c5aa5cdd620ff08a96f6b0260813d6